File: COMP_HW1/COMP_HW1_java/Family.java
import java.util.Calendar;

public class Family extends Person{
	
	private String birthday; //YYMMDD
	
	public Family(String a, String b, int c, String date) {
		super(a, b, c);
		birthday = date;
	}

	public String getBirthday() {
		return birthday;
	}

	public void setBirthday(String birthday) {
		this.birthday = birthday;
	}
	
	
	//I used the Calender class for calculating dDay()
	//Difference between current date 00:00:00(HH:MM:SS) and birthday 00:00:00
	public int dDay(){
		Calendar curr = Calendar.getInstance();
		Calendar birth = Calendar.getInstance();
		
		int year = Integer.parseInt(birthday.substring(0, 2));
		int month = Integer.parseInt(birthday.substring(2, 4));
		int date = Integer.parseInt(birthday.substring(4, 6));
		
		birth.set(1900+year, month-1, date);
		//year : YY(since 1900)
		//month : 0 - 11
		//date : 1-31
		
		//Calculating time difference between current date and birthday in milliseconds
		long difftime = curr.getTimeInMillis() - birth.getTimeInMillis();
		
		//Converting millisecond to date
		int dDay = (int)(difftime/(24 * 60 * 60 * 1000));
		
		return dDay;
	}
	
	public void print(){
		super.print();
		System.out.print("_" + birthday + "_" + this.dDay());
	}
}

File: COMP_HW1/COMP_HW1_cpp/Family.cpp
#include "Family.h"
#include <time.h>
#include <stdlib.h>
using namespace std;


Family::Family(string a, string b, int c, string birth) : Person(a, b, c){
  birthday = birth;
}

void Family::setBirthday(string birth){
  birthday = birth;
}

string Family::getBirthday(){
  return birthday;
}

//I uses <time.h> header file to calculate exact dDay
int Family::dDay(){

  //convert birthday to integer value
  int year = atoi(birthday.substr(0,2).c_str());
  int month = atoi(birthday.substr(2,2).c_str());
  int day = atoi(birthday.substr(4,2).c_str());

  time_t now;
  time (&now); //get current date

  //set birth value to indicate birthday
  struct tm *birth;
  birth = localtime(&now);
  birth->tm_year = year; //YY(after 1900)
  birth->tm_mon = month-1; //0 <= tm_mon <= 11
  birth->tm_mday = day; //1-31

  //calculate time between birthday and now in seconds
  double seconds = difftime(now, mktime(birth));
  //convert seconds to days
  int dDay = seconds / (24 * 60 * 60);

  return dDay;
}

void Family::print(){
  Person::print();
  cout << "_" << birthday << "_" << this->dDay();
}



